evalml/pipelines/pipeline_base.py

```diff
@@ -298,31 +301,31 @@ def _score_all_objectives(self, X, y, y_pred, y_pred_proba, objectives):
     def _select_y_pred_for_score(self, X, y, y_pred, y_pred_proba, objective):
         return y_pred
 
-    @classproperty
-    def model_family(cls):
+    @property
+    def model_family(self):
         """Returns model family of this pipeline template"""
-        component_graph = copy.copy(cls.component_graph)
+        component_graph = copy.copy(self._component_graph)
         if isinstance(component_graph, list):
             return handle_component_class(component_graph[-1]).model_family
         else:
-            order = ComponentGraph.generate_order(component_graph)
+            order = ComponentGraph.generate_order(component_graph.component_dict)
             final_component = order[-1]
-            return handle_component_class(component_graph[final_component][0]).model_family
+            return handle_component_class(component_graph[final_component].__class__).model_family
```

Comment:
I'm using `self._component_graph` which is the standardized ComponentGraph object, rather than `self.component_graph` which is the set one by the user so different way of accessing. I think it's quite confusing that there are these two different but very similar attributes; my understanding is that we keep it to determine if the original was a linear graph or not but for now, just decided to keep functionality as is. 😅